### DISCUSSION TITLE
post/2021-12-29-grpc-richer-error-handling/

### DISCUSSION THREAD

**Discussion by giscus** [zcong1993/blog#55] [Announcements]
# post/2021-12-29-grpc-richer-error-handling/

本文介绍 gRPC 基本错误处理和使用 goolge 扩展处理错误.

https://blog.cong.moe/post/2021-12-29-grpc-richer-error-handling/

**Comment by PeterlitsZo** [comment]
是的，无论是 HTTP status code 还是 gRPC 的 error code，本质上是不适合比较复杂的业务逻辑的。这个时候我们就不得不使用自定义的错误模型了。其实感觉底层提供的 code 最好就简单一点，标明它是正常响应还是错误就可以了，不然自己还要把业务的错误一一映射过去感觉就有点削足适履的感觉了。（就像 Rust 中的 Result 一样）

**Reply by zcong1993** [reply]
错误分为通用的错误和业务错误, 业务错误在微服务比较多的时候使用编码来确定服务域是很有必要的. 但是如果不使用最外层的状态码作为分类的话, 中间件层面上就很难通过感知错误来做一些通用化的事情, 所以不管业务错误码多么复杂还是应该用状态码分个类.
